code/datums/components/milkable.dm

```diff
@@ -0,0 +1,67 @@
+// how cows and other simplemobs are able to be milked
+
+/datum/component/milkable
+	var/mob/producer
+	var/obj/item/reagent_containers/intstorage
+	var/max_reagents
+	var/min_reagents
+	var/datum/reagent/reagent_generated
+	var/datum/callback/on_milked
+	var/list/allowed_containers = list(/obj/item/reagent_containers/glass)
+
+/datum/component/milkable/Initialize(_max_reagents, _starting_amount, _reagent_used, _callback_milked, _allowed_containers)
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	producer = parent
+
+	intstorage = new()
+	intstorage.reagents.add_reagent(_reagent_used, CLAMP(_starting_amount, 0, _max_reagents - 1))
+
+	reagent_generated = _reagent_used
+	on_milked = _callback_milked
+	if(_allowed_containers)
+		allowed_containers = _allowed_containers
+
+
+/datum/component/milkable/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/milk_animal)
+	RegisterSignal(parent, COMSIG_LIVING_REVIVE, .proc/restart_gen)
+	START_PROCESSING(SSdcs, src)
+
+/datum/component/milkable/UnregisterFromParent()
+	UnregisterSignal(parent, list(
+		COMSIG_PARENT_ATTACKBY,
+		COMSIG_LIVING_REVIVE))
+	STOP_PROCESSING(SSdcs, src)
+
+
+/datum/component/milkable/process(delta_time)
+	if(producer.stat == DEAD)
+		return PROCESS_KILL
+
+	if(producer.stat != CONSCIOUS || prob(90))
```

Comment:
Why do things need to be CONSCIOUS to produce milk, i don't think its an action that is consciously thought seems like a weird restriction

Reply:
idk but the original code for it had that check so i implemented it here